src/analyses/base.ml

```diff
@@ -753,17 +753,11 @@ struct
     | TPtr _ -> `Address (AD.bot ())
     | TComp ({cstruct=true} as ci,_) -> `Struct (bot_comp ci)
     | TComp ({cstruct=false},_) -> `Union (ValueDomain.Unions.bot ())
-    | TArray (_, None, _) -> `Array (ValueDomain.CArrays.bot ())
-    | TArray (ai, Some exp, _) -> begin
-        let default = `Array (ValueDomain.CArrays.bot ()) in
-        match eval_rv a gs st exp with
-        | `Int n -> begin
-            match ID.to_int n with
-            | Some n -> `Array (ValueDomain.CArrays.make (Int64.to_int n) (bot_value a gs st ai))
-            | _ -> default
-          end
-        | _ -> default
-      end
+    | TArray (ai, None, _) ->
+      `Array (ValueDomain.CArrays.make None (bot_value a gs st ai))
+    | TArray (ai, Some exp, _) ->
+      let l = Cil.isInteger (Cil.constFold true exp) in
```

Comment:
Why replace `eval_rv` when `Cil.constFold` does strictly less?

Reply:
Because eval_rv evaluates expression such as `x+17` for the current binding of `x`.
But this only yields the correct length if it is done in the correct place (i.e. the location of the declaration `int a[x+17];`), but we don't know where this bot_value is called.

Reply:
So this means: for VLAs the length will be `None` here (and needs to be updated later) and for normal arrays it should be the static length.

Reply:
Replaced it so the length is from `Idx`.

In `bot_value` and `init_value` we start with length `Idx.bot ()` as we are guaranteed to set the length later at some point (where the VLA is declared).

In `top_value` we use `Idx.top ()`